services/portus/Makefile

```diff
@@ -0,0 +1,146 @@
+STACK_NAME = $(notdir $(PWD))
+SWARM_HOSTS = $(shell $(DOCKER) node ls --format={{.Hostname}} 2>/dev/null)
+TEMP_COMPOSE=.stack.${STACK_NAME}.yaml
+
+# External VARIABLES
+$(if $(wildcard .env), , $(shell cp .env.config .env))
+include .env
+
+# exports
+export S3_ENDPOINT ?= $(shell hostname -I | cut -d ' ' -f1):30000
+$(info S3_ENDPOINT set to ${S3_ENDPOINT})
+MACHINE_IP = $(shell hostname -I | cut -d ' ' -f1)
+$(info MACHINE_FQDN set to ${MACHINE_FQDN})
+
+# TARGETS --------------------------------------------------
+.PHONY: help
+help: ## This colourful help
+	@awk 'BEGIN {FS = ":.*?## "} /^[a-zA-Z_-]+:.*?## / {printf "\033[36m%-20s\033[0m %s\n", $$1, $$2}' $(MAKEFILE_LIST)
+
+.PHONY: up
+up: .init .env create-certificates ${TEMP_COMPOSE}  ## Deploys graylog stack
+	$(DOCKER) stack deploy -c ${TEMP_COMPOSE} ${STACK_NAME}
+
+.PHONY: down
+down: ## Removes the stack from the swarm
+	$(DOCKER) stack rm ${STACK_NAME}
+
+.PHONY: leave
+leave: ## Leaves swarm stopping all services in it
+	-$(DOCKER) swarm leave -f
+
+.PHONY: clean
+clean: ## Cleans unversioned files
+	@git clean -dxf
+
+.PHONY: info
+info: ## expands all variables and relevant info on stack
+	$(info VARIABLES ------------)
+	$(foreach v,                                                                           \
+		$(filter-out $(PREDEFINED_VARIABLES) PREDEFINED_VARIABLES, $(sort $(.VARIABLES))), \
+		$(info $(v)=$($(v)))                                                               \
+	)
+	@echo ""
+	$(DOCKER) ps
+ifneq ($(SWARM_HOSTS), )
+	@echo ""
+	$(DOCKER) stack ls
+	@echo ""
+	-$(DOCKER) stack ps $(STACK_NAME)
+	@echo ""
+	-$(DOCKER) stack services $(STACK_NAME)
+	@echo ""
+	$(DOCKER) network ls
+endif
+
+# Helpers -------------------------------------------------
+.PHONY: .init
+.init: ## initializeds swarm cluster
+	$(if $(SWARM_HOSTS),  \
+		,                 \
+		$(DOCKER) swarm init \
+	)
+
+.PHONY: ${TEMP_COMPOSE}
+${TEMP_COMPOSE}: docker-compose.yml
+	@$(DOCKER_COMPOSE) -f $< config > $@
+	@echo "${STACK_NAME} stack file created for $@"
+
+# Self-signed authority ----------------------------------
+secrets/rootca.key:
+	@openssl genrsa -out $@ 2048
+
+secrets/rootca.crt: secrets/rootca.key
+	@read -r -p "Provide valid certificate in secrets/rootca.crt and confirm. Otherwise a self-signed certificate will be generated [y/N]: " CONTINUE; \
+    [ $$CONTINUE = "y" ] || [ $$CONTINUE = "Y" ] || \
+	(echo "Generating self-signed certificate..."; \
+	openssl req -x509 -new -nodes -key $< \
+		-subj "/C=US/ST=sparc/O=oSparc/CN=IT'IS oSparc" \
+		-sha256 -days 10000 -out $@;)
+
+install-root-certificate: secrets/rootca.crt ## installs a certificate in the host system
+	$(info installing certificate in trusted root certificates, please confirm...)
+	$(if $(or $(IS_WIN), $(IS_WSL)), \
+		-$(shell certutil.exe -user -addstore -f root $<),\
+		$(shell sudo cp $< /usr/local/share/ca-certicates/osparc.crt; sudo update-ca-certificates;))
```

Comment:
any reason to put the ca here ?

Reply:
under Windows this is necessary if one wants to use the docker client with the self-signed certificates.
under Linux I honnestly don't know yet.. I need to test it. are there better locations? I think for the docker client I can also put it in /etc/docker/certs.d/$(MACHINE_FQDN)/ca.crt

Reply:
the `/usr/local` tree is a sort of 'old style' ... I think the `/opt` tree is more appropriate for project-related stuff and then `/opt/$project`, `/etc/opt/$project`, `/var/opt/$project` accordingly